lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1749,9 +1749,22 @@ int swift::performFrontend(ArrayRef<const char *> Args,
   }
 
   int ReturnValue = 0;
-  bool HadError =
-    performCompile(*Instance, Invocation, Args, ReturnValue, observer,
-                   StatsReporter.get());
+  bool HadError;
+
+  if (Invocation.getFrontendOptions().RequestedAction ==
+      FrontendOptions::ActionType::BuildParseableInterface) {
+    const auto &InputsAndOutputs =
```

Comment:
For the sake of symmetry, I'm thinking the code would read better by factoring out then then-block here into a function so that there is symmetry with the else-half. It's also odd-seeming on the surface that performCompile handles every other action, which is how it seems to be just from looking at this change.